app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -68,29 +176,58 @@ class EventRepositoryImpl @Inject constructor(
         val createAndUpdateModifiedEvents = db.getEventDao().getModifiedEvents().filter {
             it.modifiedType == ModifiedType.CREATE || it.modifiedType == ModifiedType.UPDATE
         }.map {
-            db.getEventDao().getEventById(it.id)?.toEvent()
+            db.getEventDao().getEventById(it.id)?.toEvent(context)
         }
         createAndUpdateModifiedEvents.forEach { event ->
             event?.let {
-                val results = upsertEvent(event)
+                val (results) = upsertEvent(event)
                 if (results is Resource.Success) {
                     db.getEventDao().deleteModifiedEventById(event.id)
                 }
             }
         }
     }
 
-    override suspend fun getLocalPhotos(keys: List<String>): List<LocalPhotoDto> {
-        val keysWithJpgEnding = keys.map { it.plus(".jpg") }
-        return InternalStorage(context = context).loadImages().filter {
-            keysWithJpgEnding.contains(it.key)
-        }
+    private data class CompressEventResult(val event: AgendaItem.Event, val photosDeleted: Int)
+    private suspend fun compressEvent(event: AgendaItem.Event): CompressEventResult {
+        var photosDeleted = 0
+        val compressedEvent = event.copy(
+            photos = event.photos.mapNotNull { eventPhoto ->
+                when (eventPhoto) {
+                    is LocalPhoto -> {
+                        when (eventPhoto.savedInternally) {
+                            true -> {
+                                val byteArray = ImageStorage(context).getByteArray(eventPhoto.key)
+                                eventPhoto.copy(byteArray = byteArray)
+                            }
+                            false -> {
+                                val byteArray = eventPhoto.bitmap?.let { bitmap ->
+                                    BitmapConverters.bitmapToCompressByteArray(
+                                        bitmap, 1000000
```

Comment:
`1000000` looks a bit like a magic number, you don't know at the first glance what it stands for. I'd either use named parameters here or make it a constant, for example `MAX_PHOTO_SIZE_IN_BYTES`